app/elements/pw-shell.html

```diff
@@ -250,13 +286,44 @@
 
       // Save the initial URL so that we don't double-render the first load.
       this._currentPageURL = this._sectionRoute.path;
+
+      // Save which version of docs we're looking at, since we need to
+      // maintain this state across the app. Note that not all sections are versioned.
+      if (this._sectionData.version === '1.0' || this._sectionData.version === '2.0')
+        this.activeDocsVersion = this._sectionData.version;
+    },
+
+    _listboxVersionChanged: function(event) {
+      var newSelection = event.detail.item.textContent;
+      if (newSelection === this.activeDocsVersion)
+        return;
+      this.activeDocsVersion = newSelection;
+      this.set('_sectionRoute.path', '/' + this.activeDocsVersion + '/start/');
     },
 
     _toggleDrawer: function() {
+      // Sync the version selector with the active docs version.
+      this.$.versionSelect.selected = this.activeDocsVersion == '1.0' ? 0 : 1;
+
+      var links = this.querySelectorAll('a[data-version]');
```

Comment:
Since the contents of `<pw-shell>` is reloaded on version changed (and replaced with only links to the relevant version), this (expensive) work of toggling `hidden` should not be needed.